### ISSUE TITLE
A different value of n_batch gives different logits (llama.cpp_hf)

### ISSUE THREAD

**Issue by BadisG** [oobabooga/text-generation-webui#4478]
Hello,

I was told that n_batch doesn't alter the logits but after doing some tests I can see that it's not working as intented (or is it?).

"*When was born Tony Yoka? - Tony Yoka was born the 28 April 199...*"
The next token is supposed to be "2"
![N8IiAaQ](https://github.com/oobabooga/text-generation-webui/assets/110173477/b854ad25-c5a6-4a3c-a404-832ee14a43c3)

You can see that a low n_batch value objectively leads to a deterioration in the quality of the results (The only viable token is less likely to be chosen).

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Change the n_batch value and compare the logits.

### Logs

```shell
No logs.
```

### System Info

```shell
Windows 10.
RTX 3060.
```
